chain/consensus/compute_state.go

```diff
@@ -222,29 +224,9 @@ func (t *TipSetExecutor) ApplyBlocks(ctx context.Context,
 			return cid.Undef, cid.Undef, xerrors.Errorf("failed to serialize award params: %w", err)
 		}
 
-		rwMsg := &types.Message{
-			From:       builtin.SystemActorAddr,
-			To:         reward.Address,
-			Nonce:      uint64(epoch),
-			Value:      types.NewInt(0),
-			GasFeeCap:  types.NewInt(0),
-			GasPremium: types.NewInt(0),
-			GasLimit:   1 << 30,
-			Method:     reward.Methods.AwardBlockReward,
-			Params:     params,
-		}
-		ret, actErr := vmi.ApplyImplicitMessage(ctx, rwMsg)
-		if actErr != nil {
-			return cid.Undef, cid.Undef, xerrors.Errorf("failed to apply reward message for miner %s: %w", b.Miner, actErr)
-		}
-		if em != nil {
-			if err := em.MessageApplied(ctx, ts, rwMsg.Cid(), rwMsg, ret, true); err != nil {
-				return cid.Undef, cid.Undef, xerrors.Errorf("callback failed on reward message: %w", err)
-			}
-		}
-
-		if ret.ExitCode != 0 {
-			return cid.Undef, cid.Undef, xerrors.Errorf("reward application message failed (exit %d): %s", ret.ExitCode, ret.ActorErr)
+		rErr := t.reward(ctx, vmi, em, epoch, ts, params)
```

Comment:
I'd pass non-serialized params into this function - passing cbor-ized params into a function which in meant to be customizable feels a bit odd

Reply:
Yep, I was having a hard time deciding the best type to use as params. What would you recommend, a `cbor.Marshaler` or passing directly `reward.BlockRewardParams` and figure out if we need to generalize it more in the future? (for now, `BlockRewardParams` has all the info we need for our use cases).

Reply:
BlockRewardParams for now, can be customized later